collator/src/lib.rs

```diff
@@ -164,6 +165,31 @@ where
 			.ok()
 	}
 
+	/// Returns channels contents for each inbound HRMP channel addressed to the parachain we are
+	/// collating for.
+	///
+	/// Empty channels are also included.
+	fn retrieve_all_inbound_hrmp_channel_contents(&self, relay_parent: PHash)
+		-> Option<BTreeMap<ParaId, Vec<InboundHrmpMessage>>>
+	{
+		self
+			.polkadot_client
+			.runtime_api()
+			.inbound_hrmp_channels_contents_with_context(
+				&BlockId::hash(relay_parent),
+				sp_core::ExecutionContext::Importing,
```

Comment:
Why a special context here?

Reply:
actually, it meant to be not special (as with the dmq_contents). `Importing` seemed to be "less special" of all, at least according to [the docs](https://crates.parity.io/sp_core/enum.ExecutionContext.html). What are the other options?

Reply:
As this is part of the "block_production", I would have chosen this context. Or just no context at all :D